addons/website_sale/models/sale_order.py

```diff
@@ -483,12 +483,13 @@ def _action_confirm(self):
             phone = order.partner_shipping_id.phone
 
             # we can check if the current partner has a partner of type "delivery" that has the same address
-            existing_partner = order.env['res.partner'].search(['&', '&', '&', '&',
+            existing_partner = order.env['res.partner'].search(['&', '&', '&', '&', '&',
                                                                 ('street', '=', street),
                                                                 ('city', '=', city),
                                                                 ('state_id', '=', state),
                                                                 ('country_id', '=', country),
-                                                                ('type', '=', 'delivery')], limit=1)
+                                                                ('type', '=', 'delivery'),
+                                                                ('parent_id', '=', parent_id)], limit=1)
```

Comment:
You could reduce the diff with
```suggestion
                                                                ('parent_id', '=', parent_id),
                                                                ('type', '=', 'delivery')], limit=1)
```